src/Auth/AuthenticationCall.php

```diff
@@ -0,0 +1,9 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Auth;
+
+interface AuthenticationCall
```

Comment:
I generally don't like empty interfaces, they usually indicate a code smell of some sorts.

Reply:
We started discussing about this here: https://github.com/tempestphp/tempest-framework/pull/242#discussion_r1609698847 because every `AuthenticationCall` implementation will need different parameters and this was going to be used just to type the call parameter in the `CredentialsResolver` interface:

```php
interface CredentialsResolver
{
    public function resolve(AuthenticationCall $call): Identifiable;
}
```

How should we approach this then?